soh/soh/Enhancements/debugger/valueViewer.cpp

```diff
@@ -104,8 +109,40 @@ extern "C" void ValueViewer_Draw(GfxPrint* printer) {
     }
 }
 
+extern "C" void ValueViewer_SetupDraw() {
+    OPEN_DISPS(gGameState->gfxCtx);
+
+    Gfx* gfx;
+    Gfx* polyOpa;
+    GfxPrint printer;
+
+    polyOpa = POLY_OPA_DISP;
+    gfx = Graph_GfxPlusOne(polyOpa);
+    gSPDisplayList(OVERLAY_DISP++, gfx);
+
+    GfxPrint_Init(&printer);
+    GfxPrint_Open(&printer, gfx);
+
+    ValueViewer_Draw(&printer);
+
+    gfx = GfxPrint_Close(&printer);
+    GfxPrint_Destroy(&printer);
+
+    gSPEndDisplayList(gfx++);
+    Graph_BranchDlist(polyOpa, gfx);
+    POLY_OPA_DISP = gfx;
+
+    CLOSE_DISPS(gGameState->gfxCtx);
+}
+
+void RegisterValueViewerHooks() {
+    COND_HOOK(OnGameFrameUpdate, CVAR_VALUE, []() { ValueViewer_SetupDraw(); });
+}
+
+RegisterShipInitFunc initFunc(RegisterValueViewerHooks, { CVAR_NAME });
```

Comment:
Is this safe to do this outside of the draw loop?

Reply:
Its not outside the draw loop, `OnGameFrameUpdate` is executed before drawing is finished.

![image](https://github.com/user-attachments/assets/3214e5fb-f109-4054-a2cd-966f5f875dd6)

Reply:
perfect, appreciate the thoroughness